lib/livebook/settings.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Livebook.Settings do
+  # Keeps all Livebook settings that are backed by storage.
+  @moduledoc false
+
+  alias Livebook.FileSystem
+
+  @doc """
+  Returns the autosave path.
+
+  TODO: Make this configurable in the UI.
+  """
+  @spec autosave_path() :: String.t() | nil
+  def autosave_path() do
+    case storage().fetch_key(:settings, "global", :autosave_path) do
+      {:ok, value} -> value
+      :error -> Path.join(Livebook.Config.data_path(), "autosaved")
+    end
+  end
+
+  @doc """
+  Returns all known filesystems.
+  """
+  @spec file_systems() :: list(FileSystem.t())
+  def file_systems() do
+    [Livebook.Config.local_filesystem() | Enum.map(storage().all(:filesystem), &storage_to_fs/1)]
+  end
+
+  @doc """
+  Appends a new file system to the configured ones.
+
+  TODO: Refactor to receive settings submission parameters.
+  """
+  @spec append_file_system(FileSystem.t()) :: :ok
+  def append_file_system(%FileSystem.S3{} = file_system) do
+    attributes =
+      file_system
+      |> FileSystem.S3.to_config()
+      |> Map.to_list()
+
+    storage().insert(:filesystem, generate_filesystem_id(), [{:type, "s3"} | attributes])
```

Comment:
Shouldn't we use strings as keys? This should make the config map easier to serialize into JSON and whatnot.

Reply:
I think semantically we want them to be atoms, because they are not supposed to be dynamic. When persisting to another storage, then we will have to figure out a mechanism for them, but I would say semantically atoms is correct!